// hybrid-components/src/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importDefault(require("react"));
const styled_components_1 = __importDefault(require("styled-components"));
const styleProps_1 = require("./styleProps");
exports.Box = styled_components_1.default.div.attrs((props) => ({
    variants: props.variants || ['box']
})) `
    display: flex;
    flex-direction: column;
    box-sizing: border-box;
    width: 100%;
    height: auto;
    margin: 0;
    padding: 0;
    ${styleProps_1.variants}
    ${styleProps_1.box}
`;
exports.Box.displayName = 'Box';
exports.Card = styled_components_1.default.div.attrs((props) => ({
    variants: props.variants || ['card']
})) `
    display: flex;
    flex-direction: column;
    box-sizing: border-box;
    width: 100%;
    height: auto;
    margin: 0;
    padding: 0;
    ${styleProps_1.variants}
    ${styleProps_1.box}
`;
exports.Card.displayName = 'Card';
exports.Text = styled_components_1.default.div.attrs((props) => ({
    variants: props.variants || ['text']
})) `
    display: flex;
    flex-direction: column;
    width: 100%;
    box-sizing: border-box;
    margin: 0;
    padding: 0;
    ${styleProps_1.variants}
    ${styleProps_1.text}
`;
exports.Text.displayName = 'Text';
exports.Button = styled_components_1.default.button.attrs((props) => ({
    onClick: props.onPress,
    variants: props.variants || ['button'],
    onPress: undefined
})) `
    width: auto;
    padding: 4px 8px;
    cursor: pointer;
    display: inline-block;
    font-size: inherit;
    line-height: inherit;
    text-decoration: none;
    :active {
        opacity: 0.2;
    }
    ${styleProps_1.variants}
    ${styleProps_1.box}
`;
exports.Button.displayName = 'Button';
exports.Image = styled_components_1.default.img.attrs((props) => ({
    src: props.source,
    variants: props.variants || ['image']
})) `
    ${styleProps_1.variants}
    ${styleProps_1.box}
`;
exports.Image.displayName = 'Image';
exports.ScrollView = styled_components_1.default.div.attrs((props) => ({
    variants: props.variants || ['scrollview']
})) `
    overflow-y: scroll;
    overflow-x: hidden;
    width: 100%;
    height: auto;
    ${({ noScrollbar }) => noScrollbar
    ? `
            &::-webkit-scrollbar {
                width: 0px;
                background: transparent; /* Chrome/Safari/Webkit */
            }
            scrollbar-width: none; /* Firefox */
            -ms-overflow-style: none;  /* IE 10+ */
            `
    : ''}
    ${styleProps_1.variants}
    ${styleProps_1.box}
`;
exports.ScrollView.displayName = 'ScrollView';
exports.Hidden = styled_components_1.default.div `
    visibility: hidden;
    display: inline-block;
    width: auto;
    height: auto;
    margin: 0;
    padding: 0;
    ${styleProps_1.box}
`;
exports.Hidden.displayName = 'Hidden';
exports.Link = styled_components_1.default.a.attrs((props) => ({
    variants: props.variants || ['link']
})) `
    ${styleProps_1.variants}
    ${styleProps_1.text}
`;
exports.Link.displayName = 'Link';
exports.H1 = (props) => (react_1.default.createElement(exports.Text, Object.assign({ variants: ['h1'] }, props)));
exports.H1.displayName = 'H1';
exports.H2 = (props) => (react_1.default.createElement(exports.Text, Object.assign({ variants: ['h2'] }, props)));
exports.H2.displayName = 'H2';
exports.Row = (props) => (react_1.default.createElement(exports.Box, Object.assign({ flexDirection: 'row' }, props)));
exports.Row.displayName = 'Row';
exports.Column = exports.Box;
